src/logging.h

```diff
@@ -181,18 +257,31 @@ static inline void LogPrintf_(const std::string& logging_function, const std::st
             /* Original format string will have newline so don't add one here */
             log_msg = "Error \"" + std::string(fmterr.what()) + "\" while formatting log message: " + fmt;
         }
-        LogInstance().LogPrintStr(log_msg, logging_function, source_file, source_line);
+
+        const SourceLocation source_location{source_file, source_line};
+        LogInstance().LogPrintStr(log_msg, logging_function, source_location, skip_ratelimiting);
     }
 }
 
-#define LogPrintf(...) LogPrintf_(__func__, __FILE__, __LINE__, __VA_ARGS__)
+// Unconditional logging. Uses basic rate limiting to mitigate disk filling attacks.
+#define LogPrintf(...) LogPrintf_(__func__, __FILE__, __LINE__, /* skip_ratelimiting */ false, __VA_ARGS__)
+
+// Unconditional logging WITHOUT rate limiting. Use only for log messages that
+// MUST NOT be rate limited no matter how often they are logged. That requirement
+// should be extremely rare, so please use with care. Prefer LogPrintf(...) if
+// possible.
+#define LogPrintfWithoutRateLimiting(...) LogPrintf_(__func__, __FILE__, __LINE__, /* skip_ratelimiting */ true, __VA_ARGS__)
```

Comment:
I think it'd be nice to unify the various logprint macros by introducing new categories:

- `UNCONDITIONAL_RATE_LIMITED` replaces `LogPrintf`
- `UNCONDITIONAL_ALWAYS` replaces `LogPrintfWithoutRateLimiting`

or similar.

Reply:
Are you suggesting to get rid of `LogPrintf` completely or to define it as the following?
```c++
#define LogPrintf(...) LogPrint(UNCONDITIONAL_RATE_LIMITED, __VA_ARGS__)
```
I would not be a fan of replacing every usage of `LogPrintf` in this PR.

Reply:
I think it'd be good to get rid of `LogPrintf` completely in the future, but I'm not suggesting you do that in this PR. I'd just prefer that his PR doesn't add a new log macro.